rust/xaynet-analytics/src/database/isar.rs

```diff
@@ -39,10 +42,22 @@ impl IsarDb {
                     "failed to find all bytes from collection {}",
                     collection_name
                 )
-            });
+            })
+    }
 
-        // TODO: not sure how to proceed to parse [u8] using the collection schema. didn't find examples in Isar
-        todo!()
+    pub fn get_transaction(&self) -> Result<IsarTxn, Error> {
+        self.begin_txn(false)
+    }
+
+    pub fn get_isar_object_by_id<'txn>(
+        &self,
+        object_id: &ObjectId,
+        collection_name: &str,
+        transaction: &'txn mut IsarTxn,
+    ) -> Result<Option<IsarObject<'txn>>, Error> {
+        self.get_collection(collection_name)?
+            .get(transaction, object_id)
+            .map_err(|_| anyhow!("unable to get {:?} object", object_id))
```

Comment:
Here we cannot use `context` to keep the source error because it does not implement `Send` but we can add it at the end of the string `object ({}), object_id, e)` or something similar.

Reply:
you mean: 
`.map_err(|err| anyhow!("unable to get {:?} object ({:?})", object_id, err))` 
?

added, but now sure i understood 100%

Reply:
Yes, something like that to have more information about the original error.